src/store/client-context-provider.jsx

```diff
@@ -0,0 +1,41 @@
+import { ApolloClient, createHttpLink, InMemoryCache } from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
+import { useEffect, useState } from "react";
+import { ClientContext } from "./client-context";
+
+export const ClientContextProvider = ({ children }) => {
+  const [token, setToken] = useState(null);
+
+  useEffect(() => {
+    setToken(localStorage.getItem("jwt"));
+  }, []);
+
+  const httpLink = createHttpLink({
+    uri:
+      //in Vite, use special object `import.meta.env` to access enviroment variables
+      import.meta.env.MODE === "production"
+        ? //created environment variables must be prefixed by VITE
+          import.meta.env.VITE_API_URI
+        : "https://slopopedia-api-a5fe9aef64e8.herokuapp.com/api/graphql",
+  });
+
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        authorization: token ? `Bearer ${token}` : "",
+      },
+    };
+  });
+
+  const client = new ApolloClient({
+    link: authLink.concat(httpLink),
+    cache: new InMemoryCache(),
+  });
```

Comment:
I haven't tested extensively with `setContext` before so this may be wrong, but do we need to memoize `client` here or have some kind of `useEffect` that updates the link to use the new `authLink` whenever token gets overwritten? i.e if you login and logout a few times, does this have any kind of synchronization bugs with the `ApolloClient` instance here, or does it just re-instantiate the client each time `setToken` is called?

If it is re-instantiated each time, it's probably not a big deal for us, but it would probably be more technically sound to have a single instance initialized the first time this renders and then future renders (i.e in a `useEffect`) would update the instance instead of overwriting it